### PR TITLE
DebugUpdate

### PR DESCRIPTION
**Pull request by dschaper**
Update debug script. Verbosity and IPv6

### CODE REVIEW

File: advanced/Scripts/piholeDebug.sh [hunk 1]
```diff
@@ -330,8 +402,8 @@ function dumpPiHoleLog {
 			echo >> ${DEBUG_LOG}
 		done
 	else
-		echo "No pihole.log file found!" >> ${DEBUG_LOG}
-		printf ":::\tNo pihole.log file found!\n"
+		echo "No Pi-hole.log file found!" >> ${DEBUG_LOG}
```
Comment (PromoFaux) [inline advanced/Scripts/piholeDebug.sh]: Should be `pihole.log`

File: advanced/Scripts/piholeDebug.sh [hunk 2]
```diff
@@ -330,8 +402,8 @@ function dumpPiHoleLog {
...
+		printf ":::\tNo Pi-hole.log file found!\n"
```
Comment (PromoFaux) [inline advanced/Scripts/piholeDebug.sh]: Should also be `pihole.log`

File: advanced/Scripts/piholeDebug.sh [hunk 3]
```diff
@@ -151,6 +160,17 @@ function hostnameCheck {
...
 }
 
+function portCheck {
+    echo "############################################################" >> ${DEBUG_LOG}
+	echo "########           Open Port Information           #########" >> ${DEBUG_LOG}
+	echo "############################################################" >> ${DEBUG_LOG}
+
+    echo "::: Writing local server ports to logfile"
+
+    ${SUDO} netstat -tulpn >> ${DEBUG_LOG}
```
Comment (dschaper) [inline advanced/Scripts/piholeDebug.sh]: All ports or just 80/53?

Comment (PromoFaux) [inline advanced/Scripts/piholeDebug.sh]: Probably just 80/53. I can't see a use for any others?

Comment (diginc) [inline advanced/Scripts/piholeDebug.sh]: does `netstat` require installing netstat on any of the officially supported OSs?

I know you don't support alpine but it lacked netstat by default in my docker container and I didn't want to add time to my tests so decline installing it and instead just `curl`'d 80 and ran `dig` for port 53

Dig test: https://github.com/diginc/docker-pi-hole/blob/master/test/test_start.py#L36
Curl test: https://github.com/diginc/docker-pi-hole/blob/master/test/test_start.py#L46

Comment (dschaper) [inline advanced/Scripts/piholeDebug.sh]: We're looking for the owners of the processes at port 80 and 53 to see if they are who we expect, or if they are owned by another daemon. I guess I could change it around to run pidof or some other attempt with systemctl, but just getting a response back isn't what were looking for in this section.

There's a nslookup series of tests and I'll be adding in a couple of dig commands specific to dnsmasq to pull some metrics out of the process

Comment (dschaper) [inline advanced/Scripts/piholeDebug.sh]: Do you have `lsof`?

Comment (diginc) [inline advanced/Scripts/piholeDebug.sh]: correction, it wasn't that netstat wasn't installed it was that it was that alpine's busybox version which had quite different functionality and flags than the debian version.  LSOF seems to be same between alpine/debian.

**Comment by dschaper** [comment]
Ready for check and merge. Further updates to be applied...

**Comment by PromoFaux** [comment]
:shipit:

**Comment by dschaper** [comment]
:+1:

**Comment by dschaper** [comment]
Frozen for release.

**Comment by Mcat12** [comment]
Approved

**Pull request merged**